google/auth/external_account_authorized_user.py

```diff
@@ -154,20 +156,41 @@ def constructor_args(self):
             "token": self.token,
             "expiry": self.expiry,
             "revoke_url": self._revoke_url,
+            "scopes": self._scopes,
             "quota_project_id": self._quota_project_id,
         }
 
+    @property
+    def scopes(self):
+        """Optional[str]: The OAuth 2.0 permission scopes."""
+        return self._scopes
+
     @property
     def requires_scopes(self):
         """ False: OAuth 2.0 credentials have their scopes set when
         the initial token is requested and can not be changed."""
         return False
 
+    @property
+    def client_id(self):
+        """Optional[str]: The OAuth 2.0 client ID."""
+        return self._client_id
+
+    @property
+    def client_secret(self):
+        """Optional[str]: The OAuth 2.0 client secret."""
+        return self._client_secret
+
     @property
     def is_user(self):
         """ True: This credential always represents a user."""
         return True
 
+    @property
+    def token_info_url(self):
```

Comment:
Missing refresh_token, token_url, and audience

Reply:
Done